### ISSUE TITLE
quote-props false positive on float-like strings

### ISSUE THREAD

**Issue by silverwind** [eslint/eslint#8986]
* **ESLint Version:** 4.3.0
* **Node Version:** 8.2.1
* **npm Version:** 5.3.0

**Please show your full configuration:**
````
rules:
  quote-props: [2, as-needed]
````
**What did you do? Please include the actual source code causing the issue.**
```js
x = {'1.0': 'hey'}
```

**What did you expect to happen?**

No error.

**What actually happened? Please include the actual, raw output from ESLint.**
````
error  Unnecessarily quoted property '1.0' found  quote-props
````

Unquoting `1.0` would lead to the object key changing from `1.0` to `1` which is a breaking change to the code, so it cannot be unquoted and eslint should not recommend it.

**Comment by gyandeeps** [comment]
I cant reproduce it using online demo. Are you sure you are using eslint 4.3.0?

**Comment by silverwind** [comment]
Nevermind, I actually misinterpreted the output in my code, it ignores lines with `'1.0'`, but warns on `'1.1'` and others, which is correct.
